src/vcpkg/binarycaching.cpp

```diff
@@ -2276,6 +2568,11 @@ ExpectedL<std::vector<std::unique_ptr<IBinaryProvider>>> vcpkg::create_binary_pr
             paths, std::move(s.cos_read_prefixes), std::move(s.cos_write_prefixes)));
     }
 
+    if (s.gha_read || s.gha_write)
+    {
+        providers.push_back(std::make_unique<GHABinaryProvider>(paths, s.gha_read, s.gha_write));
```

Comment:
I think this would be a better place to check the environment variables before constructing the `GHABinaryProvider`. Then pass the tokens to the constructor. Something like the following:

```suggestion
        auto url = get_environment_variable("ACTIONS_CACHE_URL");
        auto token = get_environment_variable("ACTIONS_RUNTIME_TOKEN");
        Checks::msg_check_exit(VCPKG_LINE_INFO, (url.has_value() && token.has_value()), msgGHABinaryProvider, msg::url = "..." )
        providers.push_back(std::make_unique<GHABinaryProvider>(paths, s.gha_read, s.gha_write, url, token));
```